diff -U 3 -H -d -r -N -- xawtv-3.95~/console/fbtools.c xawtv-3.95/console/fbtools.c
--- xawtv-3.95~/console/fbtools.c	2003-02-14 15:14:04.000000000 +0100
+++ xawtv-3.95/console/fbtools.c	2008-09-01 23:17:24.000000000 +0200
@@ -21,8 +21,6 @@
 #include <linux/vt.h>
 #include <linux/fb.h>
 
-#include <asm/page.h>
-
 #include "fbtools.h"
 
 /* -------------------------------------------------------------------- */
diff -U 3 -H -d -r -N -- xawtv-3.95~/console/fbtools.h xawtv-3.95/console/fbtools.h
--- xawtv-3.95~/console/fbtools.h	2003-02-14 15:14:04.000000000 +0100
+++ xawtv-3.95/console/fbtools.h	2008-09-01 23:16:38.000000000 +0200
@@ -3,6 +3,9 @@
 #define FB_INACTIVE  2
 #define FB_ACQ_REQ   3
 
+#define PAGE_SIZE sysconf(_SC_PAGE_SIZE)
+#define PAGE_MASK (~(PAGE_SIZE - 1))
+
 /* info about videomode - yes I know, quick & dirty... */
 extern struct fb_fix_screeninfo   fb_fix;
 extern struct fb_var_screeninfo   fb_var;
diff -U 3 -H -d -r -N -- xawtv-3.95~/console/matrox.c xawtv-3.95/console/matrox.c
--- xawtv-3.95~/console/matrox.c	2005-02-03 11:40:41.000000000 +0100
+++ xawtv-3.95/console/matrox.c	2008-09-01 23:17:37.000000000 +0200
@@ -9,7 +9,6 @@
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 
-#include <asm/page.h> /* PAGE_SIZE */
 #include <linux/fb.h>
 
 #include "byteswap.h"
