diff -Nuard xawtv-3.81.orig/common/channel.c xawtv-3.81/common/channel.c
--- xawtv-3.81.orig/common/channel.c	Fri Dec  6 14:17:18 2002
+++ xawtv-3.81/common/channel.c	Sat Dec 21 19:09:34 2002
@@ -393,8 +393,13 @@
 	if (0 == cfg_parse_file(conffile))
 	    have_config = 1;
     } else {
-	sprintf(filename,"%.*s/%s",(int)sizeof(filename)-8,
-		getenv("HOME"),".xawtv");
+    
+	/* CONFIG_DIR patch */
+	if (getenv("CONFIG_DIR"))
+	    sprintf(filename,"%.*s/%.16s/%s",(int)sizeof(filename)-24,getenv("HOME"),getenv("CONFIG_DIR"),"xawtv");
+	else
+	    sprintf(filename,"%.*s/%s",(int)sizeof(filename)-8,getenv("HOME"),".xawtv");
+		
 	if (0 == cfg_parse_file(CONFIGFILE))
 	    have_config = 1;
 	if (0 == cfg_parse_file(filename))
@@ -549,8 +554,17 @@
     FILE *fp;
     int i;
 
-    sprintf(filename1,"%s/%s",getenv("HOME"),".xawtv");
-    sprintf(filename2,"%s/%s",getenv("HOME"),".xawtv~");
+    if (getenv("CONFIG_DIR"))
+	{
+	sprintf(filename1,"%s/%s/%s",getenv("HOME"),getenv("CONFIG_DIR"),"xawtv");
+	sprintf(filename2,"%s/%s/%s",getenv("HOME"),getenv("CONFIG_DIR"),"xawtv~");
+	} 
+	else 
+	{
+	sprintf(filename1,"%s/%s",getenv("HOME"),".xawtv");
+	sprintf(filename2,"%s/%s",getenv("HOME"),".xawtv~");
+	}
+	
 
     /* delete old backup */
     unlink(filename2);
--- xawtv-3.103/console/radio.c.orig	2016-07-06 21:14:56.685762659 +0200
+++ xawtv-3.103/console/radio.c	2016-07-06 21:22:24.230004771 +0200
@@ -363,6 +363,9 @@
 
     sprintf(file, "%.225s/.kde/share/config/kradiorc", getenv("HOME"));
     if (NULL == (fp = fopen(file,"r"))) {
+	if (getenv("CONFIG_DIR"))
+	    sprintf(file,"%.225s/%.16s/radio",getenv("HOME"),getenv("CONFIG_DIR"));
+	else
 	sprintf(file, "%.225s/.radio", getenv("HOME"));
 	if (NULL == (fp = fopen(file,"r")))
 	    return;
diff -Nuard xawtv-3.81.orig/console/webcam.c xawtv-3.81/console/webcam.c
--- xawtv-3.81.orig/console/webcam.c	Mon Dec  9 17:00:39 2002
+++ xawtv-3.81/console/webcam.c	Sat Dec 21 19:10:23 2002
@@ -649,7 +649,12 @@
     if (argc > 1) {
 	strcpy(filename,argv[1]);
     } else {
-	sprintf(filename,"%s/%s",getenv("HOME"),".webcamrc");
+    
+	if (getenv("CONFIG_DIR"))
+	    sprintf(filename,"%s/%s/%s",getenv("HOME"),getenv("CONFIG_DIR"),"webcamrc");
+	else
+	    sprintf(filename,"%s/%s",getenv("HOME"),".webcamrc");
+	    
     }
     fprintf(stderr,"reading config file: %s\n",filename);
     cfg_parse_file(filename);
--- xawtv-3.103/x11/rootv.c.orig	2016-07-06 21:14:56.689095974 +0200
+++ xawtv-3.103/x11/rootv.c	2016-07-06 21:26:36.498638113 +0200
@@ -9,6 +9,7 @@
  * put a TV image to the root window - requires Xvideo
  */
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -85,9 +86,12 @@
 
 static void station_list(FILE *fp)
 {
-    char filename[100];
+    char filename[PATH_MAX];
     char **list;
 
+    if (getenv("CONFIG_DIR"))
+	sprintf(filename,"%.*s/%.16s/%s",(int)sizeof(filename)-24,getenv("HOME"),getenv("CONFIG_DIR"),"xawtv");
+    else
     sprintf(filename,"%.*s/%s",(int)sizeof(filename)-8,
 	    getenv("HOME"),".xawtv");
     cfg_parse_file(CONFIGFILE);
