diff -Nur xawtv-3.23/radio/radio.c mute/radio/radio.c
--- xawtv-3.23/radio/radio.c	Wed Jul  7 22:32:44 1999
+++ mute/radio/radio.c	Sun Oct 29 22:56:48 2000
@@ -320,7 +320,27 @@
 		ifreq = fkeys[key - KEY_F(1)];
 		mvwprintw(wcommand, 1, 2, "Go to preset station %d", key - KEY_F(0));
 	    }
-	    break;	    
+	    break;
+	case 'm':
+	case 'M':
+	    if (mute == 0) {
+		mute = 1;
+	        mvwprintw(wcommand, 1, 2, "Mute");
+		radio_mute(fd);
+	    } else {
+		mute = 0;
+		radio_unmute(fd);
+	        mvwprintw(wcommand, 1, 2, "Unmute");
+	    }
+	    break;
+	case KEY_PPAGE:
+	    ifreq += 1000000;
+	    mvwprintw(wcommand, 1, 2, "Fast increment frequency");
+	    break;
+	case KEY_NPAGE:
+	    ifreq -= 1000000;
+	    mvwprintw(wcommand, 1, 2, "Fast decrease frequency");
+	    break;
 	}
     }
     if (mute)
