diff -ruN xawtv-3.81.prev/x11/atoms.c xawtv-3.81/x11/atoms.c
--- xawtv-3.81.prev/x11/atoms.c	Mon Sep  9 16:05:27 2002
+++ xawtv-3.81/x11/atoms.c	Wed Jan  1 20:45:12 2003
@@ -9,6 +9,7 @@
 Atom _NET_SUPPORTED;
 Atom _NET_WM_STATE;
 Atom _NET_WM_STATE_STAYS_ON_TOP;
+Atom _NET_WM_STATE_FULLSCREEN;
 Atom _WIN_SUPPORTING_WM_CHECK;
 Atom _WIN_LAYER;
 
@@ -60,6 +61,10 @@
     _NET_WM_STATE      = XInternAtom(dpy, "_NET_WM_STATE",    False);
     _NET_WM_STATE_STAYS_ON_TOP =
 	XInternAtom(dpy, "_NET_WM_STATE_STAYS_ON_TOP",        False);
+
+    _NET_WM_STATE_FULLSCREEN =
+	XInternAtom(dpy, "_NET_WM_STATE_FULLSCREEN",        False);
+
     _WIN_SUPPORTING_WM_CHECK =
 	XInternAtom(dpy, "_WIN_SUPPORTING_WM_CHECK",          False);
     _WIN_LAYER         = XInternAtom(dpy, "_WIN_LAYER",       False);
diff -ruN xawtv-3.81.prev/x11/atoms.h xawtv-3.81/x11/atoms.h
--- xawtv-3.81.prev/x11/atoms.h	Mon Sep  9 16:05:27 2002
+++ xawtv-3.81/x11/atoms.h	Wed Jan  1 20:46:19 2003
@@ -8,6 +8,7 @@
 extern Atom _NET_SUPPORTED;
 extern Atom _NET_WM_STATE;
 extern Atom _NET_WM_STATE_STAYS_ON_TOP;
+extern Atom _NET_WM_STATE_FULLSCREEN;
 extern Atom _WIN_SUPPORTING_WM_CHECK;
 extern Atom _WIN_LAYER;
 
diff -ruN xawtv-3.81.prev/x11/xt.c xawtv-3.81/x11/xt.c
--- xawtv-3.81.prev/x11/xt.c	Wed Oct  9 16:27:31 2002
+++ xawtv-3.81/x11/xt.c	Wed Jan  1 20:48:35 2003
@@ -809,6 +809,27 @@
 }
 #endif
 
+static void
+net_wm_state_fullscreen(Display *dpy, Window win, int set)
+{
+    XEvent e;
+
+    e.xclient.type = ClientMessage;
+    e.xclient.message_type = _NET_WM_STATE;
+    e.xclient.display = dpy;
+    e.xclient.window = win;
+    e.xclient.format = 32;
+    e.xclient.data.l[0] = set;
+    e.xclient.data.l[1] = _NET_WM_STATE_FULLSCREEN;
+    e.xclient.data.l[2] = 0;
+    e.xclient.data.l[3] = 0;
+    e.xclient.data.l[4] = 0;
+
+    XSendEvent(dpy, DefaultRootWindow(dpy), False,
+	       SubstructureRedirectMask, &e);
+}
+
+
 void
 do_fullscreen(void)
 {
@@ -827,6 +848,9 @@
     if (fs) {
 	if (debug)
 	    fprintf(stderr,"turning fs off (%dx%d+%d+%d)\n",w,h,x,y);
+	    
+	net_wm_state_fullscreen(dpy, XtWindow(app_shell), 0);
+	
 #ifdef HAVE_LIBXXF86VM
 	if (have_vm && vm_switched) {
 	    set_vidmode(vm_current);
@@ -868,6 +892,9 @@
 
 	if (debug)
 	    fprintf(stderr,"turning fs on\n");
+	    
+	net_wm_state_fullscreen(dpy, XtWindow(app_shell), 1);
+	
 	vp_x = 0;
 	vp_y = 0;
 	vp_width  = swidth;
@@ -875,6 +902,7 @@
 	XQueryPointer(dpy, RootWindowOfScreen(XtScreen(tv)),
 		      &root, &child, &rpx, &rpy, &wpx, &wpy, &mask);
 
+
 #ifdef HAVE_LIBXXF86VM
 	if (have_vm) {
 	    int i;
