diff -Nuard xawtv-3.81.orig/frequencies/Index.map xawtv-3.81/frequencies/Index.map
--- xawtv-3.81.orig/frequencies/Index.map	Fri Dec  6 12:46:59 2002
+++ xawtv-3.81/frequencies/Index.map	Sat Dec 21 18:52:20 2002
@@ -53,3 +53,6 @@
 
 [russia]
 file = secam-russia.list
+
+[cable-poland-PTK]
+file = cable-poland-PTK.list
\ No newline at end of file
diff -Nuard xawtv-3.81.orig/frequencies/cable-poland-PTK.list xawtv-3.81/frequencies/cable-poland-PTK.list
--- xawtv-3.81.orig/frequencies/cable-poland-PTK.list	Thu Jan  1 01:00:00 1970
+++ xawtv-3.81/frequencies/cable-poland-PTK.list	Sat Dec 21 18:44:50 2002
@@ -0,0 +1,260 @@
+[S-01]
+freq = 111250
+
+[S-02]
+freq = 119250
+
+[S-03]
+freq = 127250
+
+[S-04]
+freq = 135250
+
+[S-05]
+freq = 143250
+
+[S-06]
+freq = 151250
+
+[S-07]
+freq = 159250
+
+[S-08]
+freq = 167250
+
+[K-06]
+freq = 175250
+
+[K-07]
+freq = 183250
+
+[K-08]
+freq = 191250
+
+[K-09]
+freq = 199250
+
+[K-10]
+freq = 207250
+
+[K-11]
+freq = 215250
+
+[K-12]
+freq = 223250
+
+[S-09]
+freq = 231250
+
+[S-10]
+freq = 239250
+
+[S-11]
+freq = 247250
+
+[S-12]
+freq = 255250
+
+[S-13]
+freq = 263250
+
+[S-14]
+freq = 271250
+
+[S-15]
+freq = 279250
+
+[S-16]
+freq = 287250
+
+[S-17]
+freq = 295250
+
+[S-18]
+freq = 303250
+
+[S-19]
+freq = 311250
+
+[S-20]
+freq = 319250
+
+[S-21]
+freq = 327250
+
+[S-22]
+freq = 335250
+
+[S-23]
+freq = 343250
+
+[S-24]
+freq = 351250
+
+[S-25]
+freq = 359250
+
+[S-26]
+freq = 367250
+
+[S-27]
+freq = 375250
+
+[S-28]
+freq = 383250
+
+[S-29]
+freq = 391250
+
+[S-30]
+freq = 399250
+
+[S-31]
+freq = 407250
+
+[S-32]
+freq = 415250
+
+[S-33]
+freq = 423250
+
+[S-34]
+freq = 431250
+
+[S-35]
+freq = 439250
+
+[S-36]
+freq = 447250
+
+[S-37]
+freq = 455250
+
+[S-38]
+freq = 463250
+
+[K-21]
+freq = 471250
+
+[K-22]
+freq = 479250
+
+[K-23]
+freq = 487250
+
+[K-24]
+freq = 495250
+
+[K-25]
+freq = 503250
+
+[K-26]
+freq = 511250
+
+[K-27]
+freq = 519250
+
+[K-28]
+freq = 527250
+
+[K-29]
+freq = 535250
+
+[K-30]
+freq = 543250
+
+[K-31]
+freq = 551250
+
+[K-32]
+freq = 559250
+
+[K-33]
+freq = 567250
+
+[K-34]
+freq = 575250
+
+[K-35]
+freq = 583250
+
+[K-36]
+freq = 591250
+
+[K-37]
+freq = 599250
+
+[K-38]
+freq = 607250
+
+[K-39]
+freq = 615250
+
+[K-40]
+freq = 623250
+
+[K-41]
+freq = 631250
+
+[K-42]
+freq = 639250
+
+[K-43]
+freq = 647250
+
+[K-44]
+freq = 655250
+
+[K-45]
+freq = 663250
+
+[K-46]
+freq = 671250
+
+[K-47]
+freq = 679250
+
+[K-48]
+freq = 687250
+
+[K-49]
+freq = 695250
+
+[K-50]
+freq = 703250
+
+[K-51]
+freq = 711250
+
+[K-52]
+freq = 719250
+
+[K-53]
+freq = 727250
+
+[K-54]
+freq = 735250
+
+[K-55]
+freq = 743250
+
+[K-56]
+freq = 751250
+
+[K-57]
+freq = 759250
+
+[K-58]
+freq = 767250
+
+[K-59]
+freq = 775250
+
+[K-60]
+freq = 783250
+
+[K-61]
+freq = 791250
+
+[K-62]
+freq = 799250
