diff -Nru xawtv-3.94.orig/console/fbtv.c xawtv-3.94/console/fbtv.c
--- xawtv-3.94.orig/console/fbtv.c	2004-06-30 14:26:52.000000000 +0200
+++ xawtv-3.94/console/fbtv.c	2005-05-11 16:36:30.364389096 +0200
@@ -7,6 +7,10 @@
 
 #include "config.h"
 
+static char              *snapbase;
+static char              default_title[128] = "???";
+static char              message[128] = "";
+
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -187,9 +191,6 @@
 
 #define NKEYTAB (sizeof(keytab)/sizeof(struct KEYTAB))
 
-static char              *snapbase;
-static char              default_title[128] = "???";
-static char              message[128] = "";
 
 /* ---------------------------------------------------------------------- */
 /* framebuffer stuff                                                      */
Pliki xawtv-3.94.orig/console/radio i xawtv-3.94/console/radio ró¿ni± siê
Pliki xawtv-3.94.orig/console/scantv i xawtv-3.94/console/scantv ró¿ni± siê
diff -Nru xawtv-3.94.orig/console/scantv.c xawtv-3.94/console/scantv.c
--- xawtv-3.94.orig/console/scantv.c	2004-02-09 22:44:23.000000000 +0100
+++ xawtv-3.94/console/scantv.c	2005-05-11 16:32:40.264369616 +0200
@@ -16,9 +16,9 @@
 #include <pthread.h>
 
 /* xawtv */
+#include "grab-ng.h"
 #include "channel.h"
 #include "frequencies.h"
-#include "grab-ng.h"
 #include "commands.h"
 
 #include "vbi-data.h"
diff -Nru xawtv-3.94.orig/x11/vbi-gui.c xawtv-3.94/x11/vbi-gui.c
--- xawtv-3.94.orig/x11/vbi-gui.c	2003-04-14 20:06:25.000000000 +0200
+++ xawtv-3.94/x11/vbi-gui.c	2005-05-11 16:38:39.199803136 +0200
@@ -44,6 +44,7 @@
 #include "vbi-x11.h"
 #include "vbi-gui.h"
 
+#include "grab-ng.h"
 #include "channel.h"
 
 static int tt_debug = 1;
